exercises/minesweeper/test/test_minesweeper.c

```diff
@@ -0,0 +1,171 @@
+{
+   TEST_IGNORE();
+   const char *minefield[] = { "   ", "   ", "   " };
+   const char *expected[] = { "   ", "   ", "   " };
+   const size_t rows = ARRAY_SIZE(expected);
+   char **actual = annotate(minefield, rows);
+   TEST_ASSERT_EQUAL_STRING_ARRAY(expected, actual, rows);
+   free_all(actual, rows);
+}
+
+void test_annotate_minefield_with_only_mines(void)
+{
+   TEST_IGNORE();
+   const char *minefield[] = { "***", "***", "***" };
+   const char *expected[] = { "***", "***", "***" };
+   const size_t rows = ARRAY_SIZE(expected);
+   char **actual = annotate(minefield, rows);
+   TEST_ASSERT_EQUAL_STRING_ARRAY(expected, actual, rows);
+   free_all(actual, rows);
+}
+
+void test_annotate_mine_surrounded_by_spaces(void)
+{
+   TEST_IGNORE();
+   const char *minefield[] = { "   ", " * ", "   " };
+   const char *expected[] = { "111", "1*1", "111" };
+   const size_t rows = ARRAY_SIZE(expected);
+   char **actual = annotate(minefield, rows);
+   TEST_ASSERT_EQUAL_STRING_ARRAY(expected, actual, rows);
+   free_all(actual, rows);
+}
+
+void test_annotate_space_surrounded_by_mines(void)
+{
+   TEST_IGNORE();
+   const char *minefield[] = { "***", "* *", "***" };
+   const char *expected[] = { "***", "*8*", "***" };
+   const size_t rows = ARRAY_SIZE(expected);
+   char **actual = annotate(minefield, rows);
+   TEST_ASSERT_EQUAL_STRING_ARRAY(expected, actual, rows);
+   free_all(actual, rows);
+}
+
+void test_annotate_horizontal_line(void)
+{
+   TEST_IGNORE();
+   const char *minefield[] = { " * * " };
+   const char *expected[] = { "1*2*1" };
+   const size_t rows = ARRAY_SIZE(expected);
+   char **actual = annotate(minefield, rows);
+   TEST_ASSERT_EQUAL_STRING_ARRAY(expected, actual, rows);
+   free_all(actual, rows);
+}
+
+void test_annotate_horizontal_line_mines_at_edges(void)
+{
+   TEST_IGNORE();
+   const char *minefield[] = { "*   *" };
+   const char *expected[] = { "*1 1*" };
+   const size_t rows = ARRAY_SIZE(expected);
+   char **actual = annotate(minefield, rows);
+   TEST_ASSERT_EQUAL_STRING_ARRAY(expected, actual, rows);
+   free_all(actual, rows);
+}
+
+void test_annotate_vertical_line(void)
+{
+   TEST_IGNORE();
+   const char *minefield[] = { " ", "*", " ", "*", " " };
+   const char *expected[] = { "1", "*", "2", "*", "1" };
+   const size_t rows = ARRAY_SIZE(expected);
+   char **actual = annotate(minefield, rows);
+   TEST_ASSERT_EQUAL_STRING_ARRAY(expected, actual, rows);
+   free_all(actual, rows);
+}
+
+void test_annotate_vertical_line_mines_at_edges(void)
+{
+   TEST_IGNORE();
+   const char *minefield[] = { "*", " ", " ", " ", "*" };
+   const char *expected[] = { "*", "1", " ", "1", "*" };
+   const size_t rows = ARRAY_SIZE(expected);
+   char **actual = annotate(minefield, rows);
+   TEST_ASSERT_EQUAL_STRING_ARRAY(expected, actual, rows);
+   free_all(actual, rows);
+}
+
+void test_annotate_cross(void)
+{
+   TEST_IGNORE();
+   const char *minefield[] = { "  *  ", "  *  ", "*****", "  *  ", "  *  " };
```

Comment:
Can't recall if indent will allow, but for readability you could maybe format the array initialisations like:
```c
const char *minefield[] = {
        "  *  ", 
        "  *  ", 
        "*****", 
        "  *  ", 
        "  *  " 
};
```

Reply:
You are right, formatted looks much better and makes more sense as a 'minefield'.